### PR TITLE
make-containers.sh: Update images to go v1.18.1

### PR DESCRIPTION
**Pull request by javierhonduco**
For consistency with our dev Dockerfile and to unblock https://github.com/parca-dev/parca-agent/pull/341

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -36,7 +36,7 @@ RUN make build
 
 FROM ${DEBIAN_BASE} as all
 
-ARG LINUX_ARCH=x86_64
```
Comment (javierhonduco) [inline Dockerfile:39]: Same as above

File: make-containers.sh [hunk 2]
```diff
@@ -29,7 +29,6 @@ for i in "${!ARCHS[@]}"; do
         --build-arg GOLANG_BASE=$DOCKER_GOLANG_SHA \
         --build-arg DEBIAN_BASE=$DOCKER_DEBIAN_SHA \
         --build-arg ARCH=$ARCH \
-        --build-arg ARCH=$ARCH \
```
Comment (javierhonduco) [inline make-containers.sh:32]: Dup, should be a no-op

File: Dockerfile [hunk 3]
```diff
@@ -19,7 +19,7 @@ RUN apt-get -o Acquire::Check-Valid-Until="false" update -y && \
 
 WORKDIR /parca-agent
 
-ARG ARCH=amd64
```
[the remark this answers is not in the document]
Comment (kakkoyun) [reply Dockerfile:22]: It's just the default value in case if it's not specified.

[the remark this answers is not in the document]
Comment (javierhonduco) [reply Dockerfile:22]: Let's remove it to avoid confusion in case it's built-in other arches without specifying the correct one? :D

**Review by kakkoyun — APPROVED** [review]
:100:

**Pull request merged**